src/mkfs.fat.c

```diff
@@ -1635,6 +1638,15 @@ int main(int argc, char **argv)
 	    }
 	    break;
 
+    case OPT_OFFSET:
+        errno = 0;
+        part_sector = (off_t) strtoll(optarg, &tmp, 0);
```

Comment:
To prevent overflow/underflow with `off_t` type, you can use `conversion` variable and after checking for acceptable value you can assign `conversion` into `part_sector`. See usage in `case 'M':`.

Reply:
If you compile dosfstools with `--disable-largefile` then `off_t` on 32bit x86 is just 32bit. And above cast `(off_t) strtoll(...)` hides real problem.

Reply:
Should I use `uint64_t` instead of `off_t` for part_sector?

Reply:
I'm thinking about it...

`read`, `write`, `lseek`, ... handles only `off_t`. Therefore it does not make sense to use `uint64_t` for `part_sector` if `off_t` is only 32bit.

As for case `case 'M':` something like this could be used:
```c
	    errno = 0;
	    conversion = strtoll(optarg, &tmp, 0);
	    if (!*optarg || isspace(*optarg) || *tmp || errno || conversion < 0) {
		printf("Offset must be a non-negative number in sectors: %s\n", optarg);
		usage(argv[0], 1);
	    }
```

And the second part is to check if `conversion` will fit into `part_sector` which is of type `off_t`. Same problem is in pull request #114 in (code https://github.com/dosfstools/dosfstools/pull/114/files/9fcb63f4a4adac647547dcfdd1580298466375bc#diff-fc922f2c99de8cdaa0252100073322bdL481). So I would like to see this problem handled in the same way on both places. I proposed solution via `OFF_MAX`, but I'm not sure if it is correct or not.

Reply:
Fixed